test/runtests.jl

```diff
@@ -13,15 +13,30 @@ else
     typemax(Csize_t)
 end
 
+function test_path(test)
+    if test in STDLIBS
+        test_file = joinpath(STDLIB_DIR, test, "test", "runtests")
+        if !isfile(test_file * ".jl")
+            error("Standard library $test did not provide a `test/runtests.jl` file")
```

Comment:
This may be beyond the scope of this PR, but given #24628, there may or may not be a nicer way to report the error? (I didn't investigate more this question, and definitely non-blocking).